### PR TITLE
Maya: render creator has configurable options 

### PR DESCRIPTION
**Pull request by Allan-I**
## Brief description
Adds configurable render options in the settings.

## Description
Adds schema specified render options to be triggered through the settings instead of hard-coded render options in the render creator plugins.

## Additional info
The intent is to have the configurable options grabbed through a function that's within the render creator class, and that will append the settings by being triggered within the render creator function that sets the renderer settings.

### CODE REVIEW

**Comment by antirotor** [comment]
So the plan is:

- [ ] take code from #2643 as a base
- [ ] move setting schemas that define render settings from creator to its own category called `Render Settings` under Maya
- [ ] add menu item to OpenPype menu in Maya called `Set Render Settings` that will take whatever is set in the Settings and apply it
- [ ] add checkbox to `Create Render` in the Settings called `Apply Render Settings on creation` that if checked will apply render settings when Render instance is created
- [ ] connect all of that to render settings validator and its repair action, so repair action would do the same thing as `Set Render Settings` in OpenPype menu
- [ ] make validate render settings optional (or add ability to configure it so in the Settings)

File: openpype/hosts/maya/api/lib_renderproducts.py [hunk 1]
```diff
@@ -80,6 +80,13 @@
 }
 
 
+def has_tokens(string, tokens):
+    """Return whether any of tokens is in input string (case-insensitive)"""
+    pattern = "({})".format("|".join(re.escape(token) for token in tokens))
```
Comment (BigRoy) [inline openpype/hosts/maya/api/lib_renderproducts.py:85]: I believe this doesn't need to be grouped and the pattern can just be:
```suggestion
    pattern = "|".join(re.escape(token) for token in tokens)
```
```python
pattern = "|".join(re.escape(token) for token in tokens)
```

Comment (BigRoy) [reply openpype/hosts/maya/api/lib_renderproducts.py:85]: Also, is this function unused? :)

Comment (Allan-I) [reply openpype/hosts/maya/api/lib_renderproducts.py:85]: Hello Roy, thanks very much, I'm not entirely sure about this to be honest, will check it out later and let you know!

Comment (Allan-I) [reply openpype/hosts/maya/api/lib_renderproducts.py:85]: Ah, seems like it's unused, will check with the guys perhaps it's of use later on.

**Pull request closed without merging**